src/dl_typelib_write_c_header.cpp

```diff
@@ -113,6 +113,86 @@ static void dl_context_write_c_header_begin( dl_binary_writer* writer, const cha
 									   "#  if !defined(DL_STATIC_ASSERT)\n"
 									   "#    define DL_STATIC_ASSERT(x,y) // default to non-implemented.\n"
 									   "#  endif\n"
+									   "\n"
+									   "   // ... DL_ASSERT() ...\n"
+									   "#  if !defined(DL_ASSERT)\n"
+									   "#    define DL_ASSERT(x) // default to non-implemented.\n"
+									   "#  endif\n"
+									   "\n"
+									   "   // ... DLArray ...\n"
+									   "#  if defined(__cplusplus)\n"
+									   "	  template <typename T>\n"
+									   "	  struct DLArrayStruct\n"
+									   "	  {\n"
+									   "		   inline const T& operator[](uint32_t i) const\n"
+									   "		   {\n"
+									   "		       DL_ASSERT(i < count);\n"
+									   "		       return data[i];\n"
+									   "		   }\n"
+									   "		   inline T& operator[](uint32_t i)\n"
```

Comment:
I'm no fan of switching the type here... what was the reason for changing the API here?

Reply:
I will change it back. I wasn't even aware that I changed it. I based the code on the fact that the array `count` is `uint32_t` and not size_t. The "size" of the array is set to 16 bytes though, so I'm not sure what the purpose of not using size_t is?

Reply:
If you would have used pragma pack the size would have been 12... That could have been nice as well...